### PR TITLE
[full-ci] Fix archiver for protected public links

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Fixes downloading multiple resources as archive in protected public links.

## Related Issue

- Fixes https://github.com/owncloud/web/issues/6423

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

File: packages/web-app-files/src/services/archiver.ts [hunk 1]
```diff
@@ -69,8 +70,28 @@ export class ArchiverService {
...
+        },
+        responseType: 'arraybuffer'
+      })
+      const blob = new Blob([data], { type: 'application/tar' })
+      const objectUrl = URL.createObjectURL(blob)
+      const a = document.createElement('a')
+      a.download = 'download.tar'
+      a.href = objectUrl
+      a.click()
+      return objectUrl
```
Comment (kulmann) [inline packages/web-app-files/src/services/archiver.ts:91]: worth extracting this into a generic helper? Or do we even already have something generic?

File: packages/web-app-files/src/services/archiver.ts [hunk 2]
```diff
@@ -69,8 +70,28 @@ export class ArchiverService {
...
+              'Basic ' +
+              Buffer.from(['public', options.publicLinkPassword].join(':')).toString('base64')
+          })
+        },
+        responseType: 'arraybuffer'
+      })
+      const blob = new Blob([data], { type: 'application/tar' })
+      const objectUrl = URL.createObjectURL(blob)
+      const a = document.createElement('a')
+      a.download = 'download.tar'
```
Comment (kulmann) [inline packages/web-app-files/src/services/archiver.ts:88]: The archive name is supposed to be determined by the archiver service in the backend. It is not even fixed to „tar“, can also be „zip“, based on the user agent.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/32932/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32932/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2023-2-17-09-55-24.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32932/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2023-2-17-09-55-50.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32932/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-carol-2023-2-17-09-55-55.zip 


</p></details>
:boom: The e2e-oc10 tests pipeline failed. The build has been cancelled.

File: packages/web-runtime/src/plugins/web.ts [hunk 3]
```diff
@@ -65,16 +66,7 @@ export default {
...
-          a.style.display = 'none'
-          document.body.appendChild(a)
-          a.href = url
-          // use download attribute to set desired file name
-          a.setAttribute('download', name)
-          // trigger the download by simulating click
-          a.click()
-          // cleanup
-          document.body.removeChild(a)
+          triggerDownloadWithFilename(url, name)
```
Comment (kulmann) [inline packages/web-runtime/src/plugins/web.ts:69]: eh. dude... why the wrapping cascade? 😆

File: packages/web-app-files/src/services/archiver.ts [hunk 4]
```diff
@@ -111,6 +132,12 @@ export class ArchiverService {
     }
     return urlJoin(configurationManager.serverUrl, this.capability.archiver_url)
   }
+
+  private getFileNameFromResponseHeaders(headers) {
+    const contentDispositionArr = headers['content-disposition']?.split('"')
+    const fileName = contentDispositionArr[1] ?? null
+    return fileName ? decodeURI(fileName) : 'download.tar'
```
Comment (kulmann) [inline packages/web-app-files/src/services/archiver.ts:139]: Are there edge cases where the archiver service doesn't give a file name? in that case `download.tar` would be wrong on a windows machine 🙈

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/services/archiver.ts (hunk 4 above)
Comment (kulmann) [reply packages/web-app-files/src/services/archiver.ts:139]: ftr not blocking the merge.

Comment (JammingBen) [reply packages/web-app-files/src/services/archiver.ts:139]: > Are there edge cases where the archiver service doesn't give a file name?

Both serves deliver the file name via `content-disposition`. So I guess we just remove the default if it doesn't make sense anyway.

**Review by kulmann — APPROVED** [review]
🥳

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/32960/72/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUserJourney-journey1_feature-L10.png
  


</p></details>

**Pull request merged**